tests/test_checker/test_noqa.py

```diff
@@ -378,6 +378,7 @@ def test_noqa_fixture_disabled(
             '--ignore',
             ','.join(IGNORED_VIOLATIONS),
             '--disable-noqa',
+            '--disallow-union-type',
```

Comment:
We don't use configuration options for things like that.

You either requires some style, or you don't.
We only allow complexity options.

Reply:
> My way of ensuring that Union/Optional keywords are not used is by checking the name of the annotation - this will obviously not work if somebody writes the following from typing import Union as UnionType - do you have any suggestion how to solve this problem? Does similar check exist somewhere in the current code so that I could look at it?

We can probably ignore that!

> By turning this check on for the Python 3.10+ versions the project is not passing flake checks if being run on Python 3.10 interpreter - should I update all occurrences of Union and Optional in the code?

Or you can ignore this violation. It is up to you :)

Reply:
> > My way of ensuring that Union/Optional keywords are not used is by checking the name of the annotation - this will obviously not work if somebody writes the following from typing import Union as UnionType - do you have any suggestion how to solve this problem? Does similar check exist somewhere in the current code so that I could look at it?
> 
> We can probably ignore that!

OK thanks :) I know that this solution is far from ideal but I guess it will have to stay for now.

> 
> > By turning this check on for the Python 3.10+ versions the project is not passing flake checks if being run on Python 3.10 interpreter - should I update all occurrences of Union and Optional in the code?
> 
> Or you can ignore this violation. It is up to you :)

I've decided to update the codebase where it was possible - sorry for making the PR so big!